### PR TITLE
perf: MaaCore拆分任务及模板图

### PR DESCRIPTION
**Pull request by status102**
模块：
1. fight（包括各种sidestory）
2. copilot
3. credit（包括访问好友、信用购物）
4. infrast
5. award（包括每日任务、邮箱）
6. roguelike
7. 生稀盐酸
8. startup
- 其余进入base，保留在template/task/下

### CODE REVIEW

**Comment by status102** [comment]
已拆分肉鸽部分，copilot、fight、infrast等确定后一并拆分

File: src/MaaCore/Config/TaskData.cpp [hunk 1]
```diff
@@ -759,7 +841,13 @@ asst::TaskData::taskptr_t asst::TaskData::generate_match_task_info(std::string_v
     auto match_task_info_ptr = std::make_shared<MatchTaskInfo>();
     auto templ_opt = task_json.find("template");
     if (!templ_opt) {
-        match_task_info_ptr->templ_names = { std::string(name) + ".png" };
+        auto module_opt = task_json.find("module");
+        if (module_opt) {
+            match_task_info_ptr->templ_names = { (*module_opt).as_string() + "/" + std::string(name) + ".png" };
```
Comment (zzyyyl) [inline src/MaaCore/Config/TaskData.cpp:846]: 这有必要么（
干脆 templ_name 就写成带路径的不就得了（

Comment (status102) [reply src/MaaCore/Config/TaskData.cpp:846]: > 这有必要么（ 干脆 templ_name 就写成带路径的不就得了（

有些声明为empty.png、return.png的没法无脑加路径（

要是直接写在tasks.json的话，这么一搞相当于没法隐式声明了

Comment (zzyyyl) [reply src/MaaCore/Config/TaskData.cpp:846]: 隐式声明的 template 是不变的吧？比如你在 Fight/tasks.json 里声明了 Return 的 template 是 Fight/return.png，那其它文件夹的 tasks.json 用的还是这张图，也没事啊（

Comment (status102) [reply src/MaaCore/Config/TaskData.cpp:846]: 不啊，其他的会变成roguelike/return.png、infrast/return.png，就重复了

Comment (zzyyyl) [reply src/MaaCore/Config/TaskData.cpp:846]: > 不啊，其他的会变成roguelike/return.png、infrast/return.png，就重复了

你改逻辑了吗，按照现在的代码应该是不会的吧（

**Comment by status102** [comment]
模板图只是copy还没move，不知道有没有漏什么task分错了的

**Comment by ABA2396** [comment]
超级大工程）

**Comment by Constrat** [comment]
Need any help on this?

**Comment by Constrat** [comment]
> > Need any help on this?
> 
> check if tasks are classified in correct modules, thanks

None of them still need to be sorted correct?

**Comment by status102** [comment]
> None of them still need to be sorted correct?

They are loaded as one file actually. `module` field from its file name is appended to every task automatically. it's not a problem about the loading order.

I am not sure whether it's a good idea. But it looks minimally modified.

**Comment by Constrat** [comment]
> They are loaded as one file actually. `module` field from its file name is appended to every task automatically. it's not a problem about the loading order.
> 
> I am not sure whether it's a good idea. But it looks minimally modified.

My bad, what I meants is. From the original `tasks.json` are there tasks that need to be "ordered" "sorted"?

**Comment by status102** [comment]
> My bad, what I meants is. From the original `tasks.json` are there tasks that need to be "ordered" "sorted"?

not sure and didn't test

**Comment by status102** [comment]
oh a big mistask, global servers should not have all the tasks

**Comment by Constrat** [comment]
> oh a big mistask, global servers should not have all the tasks

It looks ok, I haven't seen duplicate tasks from the official

**Comment by status102** [comment]
> > oh a big mistask, global servers should not have all the tasks
> 
> It looks ok, I haven't seen duplicate tasks from the official

Oh, my browser was broken and saw wrong file, there are too many files. it looks good

**Comment by status102** [comment]
> ~It looks ok, I haven't seen duplicate tasks from the official~ ~Nvm, There's "Award" which doesn't make sense~
> 
> ~EDit: I've changed all templates ROI for Award, Recruit, Headhhunt, Friends, Infrast, Terminal, Oper so that we can remove the ROI and duplicate tasks~
> 
> ~Let me check from `dev`, then you can pull with the changed tasks.~ I'm stupid 2, I've only modified the png dimensions for official only. Nevermind what I said
> 
> They should be good as-is. I'll check the other overaseas clients

thanks a lot, busy with IELTS and makes me tired

**Comment by Constrat** [comment]
Now that I remember, with this breaking change, we need to remember to change the 
- ResourceUpdater
- Github Actions

**Review by Constrat — DISMISSED** [review]
Tasks look good for me

File: src/MaaCore/Config/TaskData.cpp
Comment (zzyyyl) [inline src/MaaCore/Config/TaskData.cpp:1]: 我超，真闲啊.jpg

Comment (status102) [reply src/MaaCore/Config/TaskData.cpp:1]: 强迫症犯了.jpg

**Comment by status102** [comment]
暂时关闭，等v5重启

**Pull request closed without merging**